seisbench/models/base.py

```diff
@@ -586,6 +579,38 @@ def _contains_callable_recursive(dict_obj):
 
         seisbench.logger.debug(f"Saved {self.name} model at {path.absolute}")
 
+    @staticmethod
+    def _get_weights_file_paths(path, version_str):
+        """
+        Return file names by parsing the path and version_str. For details, see save or load methods.
+        """
+        if isinstance(path, str):
+            path = Path(path)
+
+        if version_str is None:
+            if ".json.v" in path.name:
+                p = path.name.find(".json.v")
+                version_suffix = path.name[p + 5 :]
+            elif ".pt.v" in path.name:
+                p = path.name.find(".pt.v")
+                version_suffix = path.name[p + 3 :]
+            else:
+                version_suffix = ""
+        else:
+            version_suffix = f".v{version_str}"
```

Comment:
This follows on from the docstring commentary. 

We pass through a single path parameter, and here we are checking if the path param also includes one of the two file types (.json, .pt). If it does, we append an appropriate version suffix but we do this for both the json and pt paths. 

So, users have to specify a single file type with extension (e.g. path="mymodel.json.v1") in order to save out and load a versioned model just from the path parameter, this edge case just seems slightly non-intuitive.

Reply:
From my side, I would go with always using version strings, and then if none are provided, just defaulting back to looking up the base model with no automatic inference of the version suffix